src/LuaPrinter.ts

```diff
@@ -136,11 +147,30 @@ export class LuaPrinter {
             }
         }
 
-        this.sourceFile = path.basename(sourceFile);
+        this.sourceFile = path.basename(sourceFile, ".ts");
 
         const blockNode =  this.createSourceNode(block, this.printBlock(block));
 
-        return this.concatNodes(header, blockNode);
+        let sourceNode = this.concatNodes(header, blockNode);
+
+        if (this.options.sourceMapTraceBack) {
```

Comment:
Shouldn't this be before the main body? What if there's an error executing a global statement?

Reply:
That is a good point, the issue with that is that it modifies the source map input this function uses, but since it's only one line we can probably deal with this with a + or -1 somewhere, I'll investigate.

Reply:
That's a good point - we'll need to account for the header comment and inlined lua libs as well, won't we?